### PR TITLE
Adding all Enumerable methods

### PR DESCRIPTION
**Pull request by c-m-a**
Replicating the Enumerable Module

Methods

    my_each
    my_each_with_index
    my_select
    my_all?
    my_any?
    my_count
    my_map
    my_inject
    multiply_els
    And
    Example using multiply_els

### CODE REVIEW

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello @cma4c   🌟

**Good work on this project, but we need to fix the issues below before moving on ⬇️** 

1. **my_all?**
The method is not working properly: 

> 2.6.3 :102 > array.all?(3)
>  => false 
> 2.6.3 :103 > array.my_all?(3)
>  => true 

2. **my_any?**
That method is not well implemented as well I got the following result: 

> 2.6.3 :108 > false_arr.my_any?
>  => false 
> 2.6.3 :109 > false_arr.any?
>  => true 

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

File: lib/enumerable.rb [hunk 1]
```diff
@@ -55,6 +55,8 @@ def my_all?(pattern = nil, &block)
         return false if e.class != String
         return false if pattern.match(e).nil?
       end
+    else
+      my_each { |e| return false if e } unless pattern.nil?
```
Comment (c-m-a) [inline lib/enumerable.rb]: Add to cover the cases when there is numeric pattern [].my_all?(3), [1,3].my_all?(3)

File: lib/enumerable.rb [hunk 2]
```diff
@@ -78,7 +78,7 @@ def my_any?(pattern = nil, &block)
         return true unless pattern.match(e).nil?
       end
     else
-      my_each { |e| return true if e.nil? }
+      my_each { |e| return true if e } if pattern.nil?
```
Comment (c-m-a) [inline lib/enumerable.rb]: Add to cover this case [1].my_any?

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi @cma4c,

### Great job thus far, :raised_hands: but there are still some errors. 

- [x] When submitting a PR for review, ensure that the PR contains all changes made, not just the last change made. Also, **DO NOT** merge to the develop branch unless the code reviewer approves your code. Please fix this before your next submission.

- [x] Enumerables `my_all?` when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
`[1,2,3,4,5].my_all?(3)` should return `false`

- [x]  Enumerables `my_any?` returns true if the block never returns false or nil
`[1,2,3,4,5].my_any?{ |num| num < 10 }`  should return `true`

- [x] Enumerables `my_any?` when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
`[1,2,3,4,5].my_any?(3)` should return `true`

- [x]  Enumerables `my_none?` returns true if the block never returns false or nil
`[1,2,3,4,5].my_none?{ |num| num < 10 }` gives `error` instead of `false`

- [x]  Enumerables `my_none?` when no block or argument is given returns true when none of the collection members are false or nil
`[1,2,3,4,5].my_none?` should return `false`

- [x] Enumerables `my_none?` when a class is passed as an argument returns true if all of the collection is a member of such class
`[1,2,3,4,5].my_none?(Integer)` should return `false`

- [x] Enumerables `my_all?` when a Regex is passed as an argument returns true if all of the collection matches the Regex
`['saheed', 'oladele', 'suretrust'].my_none?(/d/)` should return `false`

- [x] Enumerables `my_none?` when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
`[1,2,3,4,5].my_none?(3)` should return `true`

- [ ] Enumerables `my_inject` combines all elements of enum by applying a binary operation, specified by a block.
`[1, 1, 1, 1, 1, 1, 1, 1].my_inject{proc { |sum, n| sum + n }}` should return `#<Proc:0x000055a055166cf0@main.rb:127>`
      
**Kindly submit for another code review once these issues have been fixed.** _Happy coding!_ :tada:

**Comment by Anna-Myzukina** [comment]
### Pull Request under the review by Anna Muzykina (TSE)

**Comment by Anna-Myzukina** [comment]
### Hi team!

### Your project looks +1 good!
#### But there are few things that need to be corrected:

Kindly recommended to check all your methods, it seems that not all of them working correctly.
Take a look at Ruby docs about Enumerable class https://ruby-doc.org/core-2.6.5/Enumerable.html

screen

### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*

**Review by Anna-Myzukina — CHANGES_REQUESTED** [review]
The project still needs changes

**Review by Redvanisation — APPROVED** [review]
# Project approved ✔️ 

### Hello team 🌟 

### _Nice work on this project, it seems that you have met the requirements so I am approving your project. 👍_ 

### You may merge to master and submit a project completion form. 🚀 

### Good luck! 🍀

**Pull request merged**